### ISSUE TITLE
YOLOv11 Output Tensor Shape Incorrect When Using 48 Classes (Output Shape 52 Instead of 53)

### ISSUE THREAD

**Issue by Khueeeee** [ultralytics/ultralytics#20522]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi Ultralytics team,

I'm encountering an issue with YOLOv11 where the output tensor shape does not match the expected number of classes.

1. Dataset: I trained a model with 48 classes (nc=48), and the dataset.yaml is correctly configured with 48 class names.
2. Training: The model trained successfully, and the logs confirm that all 48 classes are present and evaluated correctly. The final mAP and per-class metrics cover all 48 classes.
3. Model info:
Model: YOLOv11n
Ultralytics version: (before & after upgrade) – issue persists even with the latest version.
Python: 3.11
Torch: 2.6.0
4. Bug: When I inspect the output tensor using:
```
from ultralytics import YOLO
import torch

model = YOLO('/content/drive/MyDrive/Traffic/result/weights/best.pt')
out = model.model(torch.randn(1, 3, 640, 640))[0]
print(out.shape)
```

I consistently get:

`torch.Size([1, 52, 8400])`

Since the model has 48 classes, I expect:

`num_classes + 5 = 48 + 5 = 53`

5. Verification:

model.model.names returns 48 class names ✅,

All classes are present in the metrics after training ✅,

Inference & prediction on real data detect all 48 classes as expected ✅,

Only the output tensor shape is incorrect ❌.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Khueeeee, thank you for bringing this to our attention and for providing detailed context! 🚀 This is an automated response to help get you started—an Ultralytics engineer will review your issue and assist you soon.

We recommend reviewing our [Docs](https://docs.ultralytics.com/) for additional details, particularly the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many common questions are addressed there.

Since this appears to be a 🐛 Bug Report, could you please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE)? This helps us quickly track down and resolve the issue.

If you are experimenting with custom training, please include any additional information such as a snippet of your `dataset.yaml`, relevant training logs, and further details on your inference workflow.

Join the Ultralytics community in your preferred channel for more support!
- Real-time chat: [Discord](https://discord.com/invite/ultralytics) 🎧
- In-depth discussions: [Discourse](https://community.ultralytics.com/)
- Community threads: [Subreddit](https://reddit.com/r/Ultralytics)

## Upgrade

Please ensure you are running the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO models can be run in the following up-to-date, verified environments (all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) are preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your report—your detailed feedback helps us improve Ultralytics HUB and YOLO models! 🦾

**Comment by Y-T-G** [comment]
It's `num_classes + 4 = 53`. It's correct.

https://www.reddit.com/r/Ultralytics/comments/1jwl6se/ultralytics_postprocessing_guide/

**Comment by Khueeeee** [comment]
> It's `num_classes + 4 = 52`. It's correct.
> 
> https://www.reddit.com/r/Ultralytics/comments/1jwl6se/ultralytics_postprocessing_guide/

**Comment by Khueeeee** [comment]
> It's `num_classes + 4 = 52`. It's correct.
> 
> https://www.reddit.com/r/Ultralytics/comments/1jwl6se/ultralytics_postprocessing_guide/
I think
`52 = 4 (bbox) + 1 (objectness) + 47 (class scores)`

**Comment by Y-T-G** [comment]
There's no objectness score
